### PR TITLE
force package-lock.json to reference https

### PR DESCRIPTION
**Pull request by ChrisBAshton**
This is currently the advised thing to do while NPM fixes the bug: https://npm.community/t/npm-install-downgrading-resolved-packages-from-https-to-http-registry-in-package-lock-json/1818/3

Test notes: there should be no `http:` references in any `package-lock.json` file in this repository, after running an `npm install`.

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [x] Test engineer approval

### CODE REVIEW

**Review by bcmn — DISMISSED** [review]

**Review by sareh — DISMISSED** [review]
👍 Looks good

**Review by sareh — DISMISSED** [review]

**Comment by pjlee11** [comment]
[Jenkins failure](https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/psammead/detail/force-https-package-lock/4/pipeline) - when running `make install` on CI the following error is thrown:

```
> @bbc/psammead@0.1.0 force-package-lock-https /var/lib/jenkins/workspace/ad_force-https-package-lock-XOD5QFKGO7J6HJG5KVAKMLMPJNORCJAULJKDNZQLHT4D6EPR5WYQ@4

> find . -type f -name "package-lock.json" -print0 | xargs -0 sed -i '' -e 's/http:\/\//https:\/\//g'

sed: can't read : No such file or directory
```

My guess is that the difference between `npm install` locally and `npm ci` on Jenkins is causing a valid error

**Review by pjlee11 — APPROVED** [review]

**Review by pjlee11 — CHANGES_REQUESTED** [review]
Jenkins failing again, now on `sh: 1: [[: not found` following the addition of `if ! [[ -n $JENKINS_HOME ]]; then \n npm run force-package-lock-https \n fi`

**Review by bcmn — DISMISSED** [review]
Looks good & works locally. 👍  I also didn't know what `test -z` did until just now -- handy. 👀

**Review by pjlee11 — DISMISSED** [review]
Nice solution!

**Comment by bcmn** [comment]
screenshot 2018-12-14 at 11 12 09

The logging shows that this is altering node_modules? I can't imagine this'll ever be an issue, but it feels weird.

**Comment by ChrisBAshton** [comment]
argh you're right. Needs a tighter fix...

**Comment by ChrisBAshton** [comment]
Fixed!

**Review by bcmn — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -8,8 +8,10 @@
     "ci:packages": "lerna exec -- npm ci",
     "clear_styled_components": "for d in ./packages/*/*; do rm -rf $d/node_modules/styled-components; done; echo \"Cleared nested packages/**/node_modules/styled-components\"",
     "deploy-storybook": "build-storybook -o storybook_dist && gh-pages -d storybook_dist",
+    "force-package-lock-https": "find . -type f -name \"package-lock.json\" | grep -v node_modules | xargs -I % sh -c 'echo \"Replacing http->https in %\"; sed -i \"\" -e \"s/http:\\/\\//https:\\/\\//g\" %'",
     "install:packages": "lerna exec -- npm install",
     "postinstall": "npm run ci:packages",
+    "postshrinkwrap": "if test -z $JENKINS_HOME; then \n npm run force-package-lock-https \n fi",
```
Comment (dr3) [inline package.json]:
```suggestion
    "postshrinkwrap": "if test -z $CI; then \n npm run force-package-lock-https \n fi",
```
Uses https://github.com/BBC-News/psammead/blob/latest/Jenkinsfile#L22

**Review by dr3 — CHANGES_REQUESTED** [review]

**Review by sareh — APPROVED** [review]
👍

**Review by bcmn — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM

**Review by dr3 — APPROVED** [review]

**Pull request merged**